pkg/skaffold/config/options.go

```diff
@@ -94,18 +94,32 @@ type SkaffoldOptions struct {
 	WaitForDeletions WaitForDeletions
 }
 
+type RunMode string
+
+var RunModes = struct {
+	Build  RunMode
+	Dev    RunMode
+	Debug  RunMode
+	Run    RunMode
+	Deploy RunMode
+	Render RunMode
+}{
+	Build:  "build",
+	Dev:    "dev",
+	Debug:  "debug",
+	Run:    "run",
+	Deploy: "deploy",
+	Render: "render",
+}
+
 // Prune returns true iff the user did NOT specify the --no-prune flag,
 // and the user did NOT specify the --cache-artifacts flag.
 func (opts *SkaffoldOptions) Prune() bool {
 	return !opts.NoPrune && !opts.CacheArtifacts
 }
 
-func (opts *SkaffoldOptions) IsDevMode() bool {
-	return opts.Command == "dev"
-}
-
-func (opts *SkaffoldOptions) IsDebugMode() bool {
-	return opts.Command == "debug"
+func (opts *SkaffoldOptions) Mode() RunMode {
+	return RunMode(opts.Command)
```

Comment:
What should happen if we have a command with no corresponding mode?

Reply:
Are you saying that if we introduce `skaffold foo` then what would happen? I think we'd like to keep the `config.RunModes` struct up-to-date. This was the best way to do an enum that I found in Go but let me know if there's a better way.
```
var RunModes = struct {
	Build  RunMode
	Dev    RunMode
	Debug  RunMode
	Run    RunMode
	Deploy RunMode
	Render RunMode
}{
	Build:  "build",
	Dev:    "dev",
	Debug:  "debug",
	Run:    "run",
	Deploy: "deploy",
	Render: "render",
}
```

Reply:
This seems like a worthy situation to panic if it doesn't match.

Reply:
I don't think that it needs to panic. Even if the command is not found in the static `RunModes` it should still be able to function as a `RunMode` object though it might not be useful.

Reply:
For instance I've left out `init`, `completion`, `survey` commands from `RunModes` struct.

Reply:
Also `opts.Command` gets set much earlier and only allows the supported `Cobra` commands in the first place.